thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java

```diff
@@ -1190,4 +1191,30 @@ public Response getProjectedMTTD (@PathParam("autotuneId") @NotNull long id,
     double functionMTTDInHour = TimeUnit.HOURS.convert(detectionBucketSize, detectionUnit);
     return Response.ok(Math.max(functionMTTDInHour, alertFilterMTTDInHour)).build();
   }
+
+  /**
+   * Given autotuneId and a list of anomalies, return the anomaly ids that qualified for alert filter
+   * @param autotuneId
+   * @param anomalyIdsString comma split anomaly id string
+   * @return Array list of anomaly id that pass the alert filter in autotune
+   */
+  @GET
+  @Path("/eval/projected/anomalies/{autotuneId}")
+  public ArrayList<Long> getPreviewedAnomaliesByAutoTuneId (@PathParam("autotuneId") @NotNull long autotuneId,
+      @QueryParam("anomalyIds") String anomalyIdsString) {
+    String[] anomalyIds = anomalyIdsString.split(COMMA_SEPARATOR);
+    //Initiate tuned alert filter
+    AutotuneConfigDTO target = DAO_REGISTRY.getAutotuneConfigDAO().findById(autotuneId);
+    Map<String, String> tunedParams = target.getConfiguration();
+    BaseAlertFilter alertFilter = alertFilterFactory.fromSpec(tunedParams);
+    ArrayList<Long> idList = new ArrayList<>();
+    for (String anomalyId : anomalyIds) {
+      Long id = Long.valueOf(anomalyId);
```

Comment:
Make id a long type instead of Long to elimiate the possibility of assigning it with null accidentally.